tests/interpreter/cases/call/or.yaml

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+cases:
+  - note: or-error
+    data: {}
+    modules:
+      - |
+        package test
+
+        inc(x) = x + 1
+        inc(x) = x + 4
+
+        a1 = inc(5)
+    query: data.test
+    error: functions must not produce multiple outputs for same inputs
+
+  - note: or-ok
+    data: {}
+    modules:
+      - |
+        package test
+
+        inc(x) = x + 1
+        inc(x) = x + 1
+
+        a1 = inc(5)
+    query: data.test
+    want_result:
+      a1: 6
+
+  - note: or-true
+    data: {}
+    modules:
+      - |
+        package test
+
+        inc(x) = 7 == x
+        inc(x) = 7 == x
+
+        a1 = inc(7)
+    query: data.test
+    want_result:
+      a1: true
+
+  - note: or-false
+    data: {}
+    modules:
+      - |
+        package test
+
+        inc(x) = 8 == x
+        inc(x) = 3 == x
+
+        a1 = inc(2)
+    query: data.test
+    want_result:
+      a1: false
+
+  - note: or-undefined
+    data: {}
+    modules:
+      - |
+        package test
+
+        inc(x) = x + 1
+        inc(x) = x + 1
+        inc(x) = y {
+           x > 10 # This will evaluate to false.
+           y = 100 # y will be undefined.
+        }
+
+        a1 = inc(5)
+    query: data.test
+    want_result:
+      a1: 6
+      
```

Comment:
Add test cases for the following scenarios:
1, First definition produces undefined value.
2. All definitions produce undefined value.

Confirm expected behavior in Rego playground.

Reply:
I changed a bit my approach to cover all the cases